onedal/primitives/tree_visitor.cpp

```diff
@@ -75,14 +75,45 @@ class skl_tree_node {
 // We only expose the minimum information to python
 template <typename T>
```

Comment:
What's the difference between `tree_state` and `tree_state_py` ? Is the former still required after your changes?

Reply:
Ahh, so I added tree_state_py which matches behavior in daal4py's gettree: https://github.com/intel/scikit-learn-intelex/blob/master/src/gettree.pyx#L58 . The pyTreeState is used in a weird location: https://github.com/intel/scikit-learn-intelex/blob/master/generator/wrapper_gen.py#L1259 but does it at the very end.  The changes are related to the py::array_t only be used at the very end (thus internally only c++ data structures are used).

Reply:
I didn't notice the subtle differences between in the data members `tree_state` and `tree_state_py`. Looks are both a justified.